### DISCUSSION TITLE
create uniform mesh

### DISCUSSION THREAD

**Discussion by shivamsx** [pyvista/pyvista#2624] [Q&A]
I am having a 3D assembly in .stl format but it is not meshed uniformly . I want to update the mesh in to it , Existing meshes are very small in size I want to find centroid of all components in that assembly so I want to make my mesh uniform how it can be done please answer me..

**Comment by akaszynski** [comment]
Hello,

I'd take a look at [pyacvd](https://github.com/pyvista/pyacvd) and try that.

**Comment by shivamsx** [comment]
[assembly.zip](https://github.com/pyvista/pyvista-support/files/7218913/assembly.zip)

Here is my code.
```py
import pyvista as pv
import pyacvd
filename = '/Users/shivam/AppData/Local/pyvista/pyvista/examples/assembly.stl'
mesh = pv.read(filename)
p = pv.Plotter()
p.add_mesh(mesh, show_edges = True, edge_color= 'black', color = 'yellow',smooth_shading= True)
p.set_background('lavender', all_renderers=False)
p.show()
mesh.center
```

**Comment by shivamsx** [comment]
Hello is there any solution for the issue which I have raised above akaszynski  please let me know if there is any solution for that.
